File: contents/euclidean_algorithm/code/javascript/euclidean_example.js
function euclidMod(a, b) {
  a = Math.abs(a);
  b = Math.abs(b);

  let temp;
  while (b !== 0) {
    temp = b;
    b = a % b;
    a = temp;
  }

  return a;
}

function euclidSub(a, b) {
  a = Math.abs(a);
  b = Math.abs(b);

  while (a !== b) {
    if (a > b) {
      a -= b;
    } else {
      b -= a;
    }
  }

  return a;
}

console.log('[#]\nModulus-based euclidean algorithm result:')
console.log(euclidMod(64 * 67, 64 * 81));
console.log('[#]\nSubtraction-based euclidean algorithm result:')
console.log(euclidSub(128 * 12, 128 * 77));

File: contents/euclidean_algorithm/code/perl/euclidean_example.pl
#!/usr/bin/env perl
use strict;
use warnings;

sub euclid_sub {
    my ($a , $b) = @_ ;
    $a = abs( $a );
    $b = abs( $b );

    while ( $a != $b ) {
        if ( $a > $b ) {
            $a = $a - $b;
        } else {
            $b = $b - $a;
        }
    }

    return $a;
}

sub euclid_mod {
    my ($a , $b) = @_;
    $a = abs( $a );
    $b = abs( $b );

    while ( $b != 0 ) {
        my $temp = $b;
        $b = $a % $b;
        $a = $temp;
    }

    return $a;
}

sub main {
    $check1 = euclid_mod(64*67,64*81);
    $check2 = euclid_sub(128*12,128*77);
    print $check1."\n";
    print $check2."\n";
}

main()
